File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.cs
public class Solution {
    public int MinimumBoxes(int[] apple, int[] capacity) {
        int tsum = apple.Sum();
        Array.Sort(capacity);
        Array.Reverse(capacity);

        int count = 0;
        while (tsum > 0) {
            tsum -= capacity[count];
            count += 1;
        }

        return count;
    }
}

File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        tsum = sum(apple)

        if tsum == sum(capacity):
            return len(capacity)

        capacity.sort(reverse=True)
        count = 0

        while tsum > 0:
            tsum -= capacity[count]
            count += 1

        return count
